easybuild/easyblocks/c/cuda.py

```diff
@@ -78,6 +91,35 @@ def install_step(self):
         #overriding maxhits default value to 300 (300s wait for nothing to change in the output without seeing a known question)
         run_cmd_qa(cmd, qanda, std_qa=stdqa, no_qa=noqanda, log_all=True, simple=True, maxhits=300)
 
+
+    def post_install_step(self):
+        # Wrapper script definition
+        wrapper = """#!/bin/sh
+echo "$@" | grep -e '-ccbin' -e '--compiler-bindir' > /dev/null
+if [ $? -eq 0 ];
+then
+        echo "ERROR: do not set -ccbin or --compiler-bindir when using the `basename $0` wrapper"
+else
+        nvcc -ccbin=%s "$@"
+        exit $?
+fi """
+
+        def create_wrapper(wrapper_name,wrapper_comp):
+	    wrapper_f = os.fdopen(os.open("%s/bin/%s" % (self.installdir, wrapper_name), os.O_WRONLY | os.O_CREAT , stat.S_IXUSR | stat.S_IRUSR | stat.S_IXGRP | stat.S_IRGRP | stat.S_IXOTH | stat.S_IROTH ),'w')
+	    wrapper_f.write(wrapper % wrapper_comp)
+	    wrapper_f.close()
+
+	# Prepare wrappers to handle a default host compiler other than g++
+        if self.cfg['generate_intel_wrapper']:
+            wrapper_name = "invcc"
+            wrapper_comp = "icpc"
```

Comment:
Why not always include both wrappers (or maybe additional ones in the future, e.g. for PGI)?

So, only define `generate_wrappers` as an extra easyconfig parameter, and go:

```python
if self.cfg['generate_wrappers']:
    wrappers = [
        ('gnvcc', 'g++'),
        ('invcc', 'icpc'),
    ]
    for wrapper_name, wrapper_comp in wrappers:
        create_wrapper(wrapper_name, wrapper_comp)
```

If deemed necessary, you can only define the wrapper that make sense, by checking via the `which` function whether the wrapper compiler is available, and only then defining the matching wrapper.